src/rgw/rgw_rest_s3.cc

```diff
@@ -684,7 +684,10 @@ void RGWListBucket_ObjStore_S3::send_versioned_response()
   s->formatter->dump_string("VersionIdMarker", marker.instance);
   if (is_truncated && !next_marker.empty()) {
     s->formatter->dump_string("NextKeyMarker", next_marker.name);
-    s->formatter->dump_string("NextVersionIdMarker", next_marker.instance);
+    if (next_marker.instance.empty())
+      s->formatter->dump_string("NextVersionIdMarker", "null");  
```

Comment:
I think keep it as a empty string is OK.

Think about the using case that someone get "NextVersionIdMarker" field, then decide whether use this field or not by judging "NextVersionIdMarker.empty()".

Reply:
@ZVampirEM77 That's right. It will never be empty now.

Reply:
@shashalu @joscollin Maybe it is my bad. I find S3 really returns "null" when some field is empty.

eg:
When I list object versions, if the bucket versioning function is disable, the response msg likes following:
{u'Name': 'XXX', 'ResponseMetadata': {'HTTPStatusCode': 200, 'RetryAttempts': 0, 
...
u'Versions': [{
...
u'VersionId': 'null', 
...
, u'Prefix': '', u'KeyMarker': '', u'IsTruncated': False, u'VersionIdMarker': ''}

Yeah, the "VersionId" is set to "null".

@shashalu  So could you try to confirm what will S3 return if the "NextVersionIdMarker" is empty. In my mind, the response msg of S3 will not include the "NextVersionIdMarker" field when "NextVersionIdMarker" is empty. Is it right?

Reply:
I don't know what will S3 return.
I modify this just want to solve the following case.

There are three objects in a bucket.
objecta1 VersionId=1
objecta1 VersionId=null
objecta2 VersionId=1

list_bucket_versions with max-keys=2, original response will be "NextKeyMarker = objecta1; NextVersionIdMaker is empty"
Then use the NextKeyMarker and NextVersionIdMarker to list the remain objects in this bucket.
The response will be start with objecta1,VersionId=1. Actually, the expected response is start with objecta2,VersionId=1.

If using this "NextKeyMarker = objecta1; NextVersionIdMaker=null" to list the remain objects, the response is correct.

Reply:
@shashalu For keeping compatible with S3, I think we should have a consistent response msg here.